core/controllers/profile.py

```diff
@@ -364,3 +368,28 @@ def put(self):
         user_services.update_preferred_site_language_code(
             self.user_id, site_language_code)
         self.render_json({})
+
+
+class UserInfoHandler(base.BaseHandler):
+    """Provides info about user."""
+
+    @acl_decorators.require_user_id_else_redirect_to_homepage
```

Comment:
This decorator seems like an odd choice. This is an AJAX handler, right? So why would redirection happen?

Please also add backend tests for the cases where a user accesses this handler and (i) is logged in, (ii) is not logged in. If this handler is not meant to be called for logged-out users, please make sure it raises a 400 error in that case.

Reply:
I have created a new decorator that will return 401 when user_id is not available (=user is not logged in).